### PR TITLE
Allow referencing packages not under the directory containing pnpm-workspace.yaml

### PR DESCRIPTION
**Pull request by D4N14L**

### CODE REVIEW

**Comment by zkochan** [comment]
I guess, rewrite the failing test: `shared-workspace-lockfile: don't install dependencies in projects that are outside of the current workspace`

**Comment by zkochan** [comment]
Seems like there's a conflict with this and #1451.

What if one of the packages is a symlink. And the real location of that package is not matched by the globs?

File: packages/plugin-commands-installation/src/recursive.ts [hunk 1]
```diff
@@ -420,6 +421,25 @@ async function readLocalConfig (prefix: string) {
...
 
+function calculateRepositoryRoot (
+  workspaceDir: string,
+  projectDirs: string[]
+) {
+  // assume repo root is workspace dir
+  let relativeRepoRoot = '.'
+  for (const rootDir of projectDirs) {
+    const relativePartRegExp = new RegExp(`^(\\.\\.\\${path.sep})+`)
+    const relativePartMatch = relativePartRegExp.exec(path.relative(workspaceDir, rootDir))
```
Comment (zkochan) [inline packages/plugin-commands-installation/src/recursive.ts:432]: Is a regex really needed here? Seems like a simle **.startsWith(\`..${path.sep}\`)** would be enough.

**Comment by zkochan** [comment]
@D4N14L is it fine if I release it only in pnpm v5?

**Comment by D4N14L** [comment]
@zkochan Releasing in v4 would also be appreciated as this is what we’re using internally. If you would like me to cherry-pick the change into the v4 branch I can do that.

**Comment by zkochan** [comment]
ok, I will do it

**Comment by zkochan** [comment]
:ship: 4.14.3

**Pull request merged**